runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -729,6 +879,42 @@ TR_VectorAPIExpansion::getDataTypeFromClassNode(TR::Compilation *comp, TR::Node
    return fej9->getClassPrimitiveDataType(clazz);
    }
 
+TR_OpaqueClassBlock *
+TR_VectorAPIExpansion::getArrayClassFromDataType(TR::Compilation *comp, TR::DataType type)
+   {
+   TR_J9VMBase *fej9 = comp->fej9();
+   J9JavaVM *vm = fej9->getJ9JITConfig()->javaVM;
+   J9Class *j9class;
+
+   switch (type)
+      {
+      case TR::Float:
+         j9class = vm->floatArrayClass;
+         break;
+      case TR::Double:
+         j9class = vm->doubleArrayClass;
+         break;
+      case TR::Int8:
+         j9class = vm->byteArrayClass;
+         break;
+      case TR::Int16:
+         j9class = vm->charArrayClass;
```

Comment:
Is there any reason for using `charArrayClass` instead of `shortArrayClass`?
All other cases with `TR::Int` in this switch statement use array classes of signed integer types.

Reply:
You are right, I think we need `shortArrayClass`. Fixed.